chcemvediet/apps/inforequests/admin.py

```diff
@@ -269,8 +321,16 @@ def get_queryset(self, request):
         queryset = queryset.select_related(u'advanced_by')
         return queryset
 
+    def get_inforequest(self, obj):
+        return obj.inforequest
+
+    def has_delete_permission(self, request, obj=None):
+        if obj is None:
+            return True
+        return not obj.is_main
+
 @admin.register(Action, site=admin.site)
-class ActionAdmin(admin.ModelAdmin):
+class ActionAdmin(NoBulkDeleteAdminMixin, DeleteNestedInforequestEmailAdminMixin):
```

Comment:
Pls pri pouzivani mixinov vzdy nechaj na konci aj non-mixin model od ktoreho dedis. Tj. teraz `admin.ModelAdmin`. To nevadi, ze classa je podedena od `admin.ModelAdmin` viacerymi cestami. Python to pochopi. Len mixiny musia byt pred hlavnou classou.

Reply:
:ok: